src/prebuilt/wasm2c.include.c

```diff
@@ -32,6 +32,37 @@ const char SECTION_NAME(declarations)[] =
 "  if (UNLIKELY((a) + sizeof(t) > mem->size)) TRAP(OOB)\n"
 "#endif\n"
 "\n"
+"#if WABT_BIG_ENDIAN\n"
+"static inline void load_data(void *dest, const void *src, size_t n) {\n"
+"  size_t i = 0;\n"
+"  u8 *dest_chars = dest;\n"
+"  memcpy(dest, src, n);\n"
+"  for (i = 0; i < (n>>1); i++) {\n"
+"    u8 cursor = dest_chars[i];\n"
+"    dest_chars[i] = dest_chars[n - i - 1];\n"
+"    dest_chars[n - i - 1] = cursor;\n"
+"  }\n"
+"}\n"
+"#define LOAD_DATA(m, o, i, s) load_data(&(m.data[m.size - o - s]), i, s)\n"
+"#define DEFINE_LOAD(name, t1, t2, t3)                                                 \\\n"
+"  static inline t3 name(wasm_rt_memory_t* mem, u64 addr) {                            \\\n"
+"    MEMCHECK(mem, addr, t1);                                                          \\\n"
+"    t1 result;                                                                        \\\n"
+"    __builtin_memcpy(&result, &mem->data[mem->size - addr - sizeof(t1)], sizeof(t1)); \\\n"
```

Comment:
@SoniEx2 @binji Maybe I'm missing something here, but is there any particular reason why the memory is addressed like this, instead of addressing it with `&mem->data[addr]` like in the little-endian?

Also, related, is there any way to make the `load_*` functions accessible?

Reply:
This avoids a byteswap. Whether it has any effects on performance on s390x specifically is questionable, but it can make a difference on other big-endian platforms.

And, the `load_*` functions?